wsmaster/che-core-api-factory-github-common/src/main/java/org/eclipse/che/api/factory/server/github/AbstractGithubURLParser.java

```diff
@@ -78,6 +79,7 @@ public abstract class AbstractGithubURLParser {
     this.apiClient = githubApiClient;
     this.disableSubdomainIsolation = disableSubdomainIsolation;
     this.providerName = providerName;
+    this.isGitHubServer = !isNullOrEmpty(oauthEndpoint);
```

Comment:
why if there is no OAuth endpoint we treat is as GitHub server?

Reply:
If the `oauth endpoint` is not set, it means that the provider is GitHub SAAS by default. Otherwise we treat it as GitHub Server

Reply:
@vinokurig can we possibly extract this logic to the separate method and add javdoc clarity?